block_cascade/cli/crud.py

```diff
@@ -0,0 +1,259 @@
+
+    # CreatePersistentResoure SDK
+    region = resource.environment.region
+    client_options = {"api_endpoint": f"{region}-{SERVICE}"}
+    client = aiplatform.PersistentResourceServiceClient(client_options=client_options)
+
+    parent = get_parent_str(resource)
+
+    try:
+        operation = client.create_persistent_resource(
+            parent=parent,
+            persistent_resource_id=resource.persistent_resource_id,
+            persistent_resource=persistent_resource,
+        )
+        click.echo(
+            """
+            Attempting to create Persistent Resource... 
+            This may take several minutes and will continue if you close this terminal.
+            """
+        )
+        response = operation.result()
+        response = json.loads(json_format.MessageToJson(response._pb))
+        click.echo(f'Persistent resource {response["name"]} created successfully.')
+    except Exception as e:
+        click.echo(e)
+
+
+@click.command()
+@click.option(
+    "--project",
+    "-p",
+    help="GCP project ID. Inferred from gcloud config if not provided.",
+)
+@click.option(
+    "--region",
+    "-r",
+    help="GCP region. Inferred from gcloud config if not provided.",
+)
+def list_persistent_resources(region: str = None, project: str = None):
+    # try to infer project and region from gcloud config
+    if region is None or project is None:
+        gcp_config = get_gcloud_config()
+        if project is None:
+            project = get_gcp_project(gcp_config)
+        if region is None:
+            region = get_gcp_region(gcp_config)
+
+    client_options = {"api_endpoint": f"{region}-{SERVICE}"}
+    client = aiplatform.PersistentResourceServiceClient(client_options=client_options)
+
+    click.echo("Querying the list of Persistent Resources...")
+    request = aiplatform.ListPersistentResourcesRequest(
+        parent=f"projects/{project}/locations/{region}"
+    )
+
+    response = client.list_persistent_resources(request=request)
+    response_json = json.loads(json_format.MessageToJson(response._pb))
+    click.echo("Listing Persistent Resources:\n")
+    click.echo(json.dumps(response_json, indent=2))
+
+
+@click.command()
+@click.option(
+    "--persistent-resource-id",
+    "-i",
+    required=True,
+    help="ID of the persistent resource to delete.",
+)
+@click.option(
+    "--project",
+    "-p",
+    help="GCP project ID. Inferred from gcloud config if not provided.",
+)
+@click.option(
+    "--region",
+    "-r",
+    help="GCP region. Inferred from gcloud config if not provided.",
+)
+def delete_persistent_resource(
+    persistent_resource_id: str, region: str = None, project: str = None
+):
+    # try to infer project and region from gcloud config
+    if region is None or project is None:
+        gcp_config = get_gcloud_config()
+        if project is None:
+            project = get_gcp_project(gcp_config)
+        if region is None:
+            region = get_gcp_region(gcp_config)
+
```

Comment:
Should exit early if `project` and/or `region` cannot be read from `gcloud` configuration.